rename "Visual validation" to "snapshot".

cadeParade opened the pull request:
##  [Clubhouse Card](https://app.clubhouse.io/percy/story/2191/rename-visual-validation-to-snapshot)

## Description
Rename all instances of "visual validation" to "snapshot". Also add an asterisk with a tooltip after each instance of this term to define it.

## Pictures
Tooltip:
![snapshot-asterisks](https://user-images.githubusercontent.com/3179218/42536685-932bc0fc-8447-11e8-9bae-7582576558ed.gif)

Build Info dropdown
![build__3934_-_percy_percy-web](https://user-images.githubusercontent.com/3179218/42535386-14613804-8444-11e8-81db-5b329d27eddb.png)

Billing page:
![percy___continuous_visual_reviews_for_web_apps](https://user-images.githubusercontent.com/3179218/42535499-5c047e6e-8444-11e8-9134-de3f7354fe20.png)

Pricing page:
![pricing_-_percy___continuous_visual_reviews_for_web_apps](https://user-images.githubusercontent.com/3179218/42536114-070ceebc-8446-11e8-9580-3e3a61a06672.png)

Index:
![percy___continuous_visual_reviews_for_web_apps](https://user-images.githubusercontent.com/3179218/42536356-ac415544-8446-11e8-805c-56998fe9701d.png)



## Reviewing this PR
How do we feel about the ui of this? Too repetetive?
_**Is Covered by Automated Tests?: NO**_
Hover states are notoriously difficult to reproduce in tests.

timhaines commented on app/templates/components/pricing-questions.hbs:
```diff
@@ -1,21 +1,21 @@
+    <h4>What is a snapshot?</h4>
     <p>
-      Each full-page render and visual diff combined counts as a single visual validation.
+      Each full-page render and visual diff combined counts as a single snapshot.
```
Maybe `Each screenshot and visual diff..` --  to be inclusive of component screenshots.

timhaines replied:
(acknowledging this is unrelated to your change - but perhaps an appropriate edit to make while we're here)

fotinakis replied:
Need to remove "full-page" from here as well.

PixelJanitor commented:
Didn't we just do this rename recently throughout the site/app?

timhaines commented on app/templates/components/snapshot-explanation-tooltip.hbs:
```diff
@@ -0,0 +1 @@
+<div class="hint--top hint--rounded hint--bounce" aria-label="Snapshot usage includes all screenshots rendered by Percy, including each width and browser.">*</div>
```
Suggestion for wording - not 100% sold on either option: 

"Snapshot usage includes all screenshots rendered `for your builds, inclusive of` each width and browser."   

The words in `backticks` are just to highlight the changes.

fotinakis replied:
I like the "by/in Percy" in the original, to remind that we're rendering them. I don't think "builds" is a necessary qualifier here.

I've created a doc with some ideas so we can iterate on this and finalize it:
https://docs.google.com/document/d/1EOkKSaHOuQegvuaWHArUZxTxbZDM6HqKMyNWc6qM-DQ/edit

timhaines commented:
@PixelJanitor the change to visual validations was quite a long time ago now (in startup years).  I submitted a PR to change it to screenshots, but we abandoned it in favor of snapshots (this PR).

PixelJanitor commented:
Ah yes - @timhaines this is what I was thinking of https://github.com/percy/percy-web/pull/442

timhaines commented:
@cadeParade I don't think the tooltip is too repetitive - it only looks that way in your gif - normal users won't hover it that way. :)

However, I think we should have the text displayed on the pricing page somewhere staticly, so it doesn't look like (or can be interpreted as though) it's hidden/hard to find.  We can still do the tooltips, but having the language present on the page (additionally) feels pretty important to me.

timhaines commented:
@cadeParade @fotinakis I'm also wondering if we should add 2 rows to the build info dropdown to ease understanding and transparency:
* Total Snapshots: [Total number of comparisons.  i.e. 400]
* Consolidated Snapshots: [Total number of snapshot records.  i.e. 100]

fotinakis commented on app/templates/index.hbs:
```diff
@@ -83,7 +83,7 @@
       {{inline-svg "responsive"}}
     </figure>
-    <h3 class="text-xl md:text-2xl mb-1">Responsive visual validations</h3>
+    <h3 class="text-xl md:text-2xl mb-1">Responsive snapshots {{snapshot-explanation-tooltip}}</h3>
```
This shouldn't be changed on this marketing page. This is talking about the value of Responsive visual validations, not our pricing.

cadeParade replied:
Sorry, I think I'm missing some context. So then we'll have references to "visual validations" and "snapshot" which are both referring to the same thing? I thought we were removing "visual validation" language because customers were often confused about its meaning...

timhaines replied:
Yeah, I'm +1 on removing visual validations here too.  Would `Responsive width snapshots` or `Responsive screenshots` work better?

The tooltip shouldn't be in this location.

cadeParade replied:
`Responsive visual testing`?

cadeParade commented:
Do we like asterisk or info icon better? @PixelJanitor  @timhaines @paytonealexis 
<img width="225" alt="percy___continuous_visual_reviews_for_web_apps" src="https://user-images.githubusercontent.com/3179218/42485891-7c26a1ae-83ae-11e8-88e2-89ba0117f861.png">

fotinakis commented on app/templates/components/pricing-section.hbs:
```diff
@@ -56,8 +56,8 @@ Most Percy builds finish in under 2 minutes.">
-    <div class="PricingSection-diffs">{{format-number bucket.plan.numDiffs}} visual validations included</div>
-    <div class="PricingSection-infoItem">${{bucket.plan.extraDiffPrice}} / extra visual validation</div>
+    <div class="PricingSection-diffs">{{format-number bucket.plan.numDiffs}} snapshots {{snapshot-explanation-tooltip}} included</div>
+    <div class="PricingSection-infoItem">${{bucket.plan.extraDiffPrice}} / extra snapshot {{snapshot-explanation-tooltip}}</div>
```
Don't think we need it here, we don't need it on every single variation (esp. since this is right under one with the asterisk)

fotinakis commented on app/templates/components/organizations/billing-section.hbs:
```diff
@@ -187,7 +187,7 @@
                   </div>
                   <div>
-                    Unlimited visual validations
+                    Unlimited snapshots {{snapshot-explanation-tooltip}}
```
Can you change this to "Custom snapshots" please.

cadeParade replied:
Sure, but why is this "custom snapshots" and none of the others are?

timhaines replied:
@cadeParade I think Mike's intent here is to get away from the word 'Unlimited'.  Each other plan listed has a number of snapshots.  Custom is consistent with the wording for other items in the enterprise column..

cadeParade replied:
I see, so "Custom" takes the place of "Unlimited"?

timhaines replied:
@cadeParade  Yes - in this one particular instance.  `Unlimited visual validations` -> `Custom snapshots`

cadeParade replied:
Ok, that makes more sense. Thanks.

fotinakis commented on app/templates/components/organizations/billing-section.hbs:
```diff
@@ -83,7 +83,7 @@
           {{#if organization.subscription.plan.isFree}}
             <div class="Alert Alert--warning">
-              <strong>Your trial has expired.</strong> You're currently on the limited-use free plan, which provides {{subscriptionData.PLANS.[0].numWorkersTitle}} and {{subscriptionData.PLANS.[0].numDiffs}} visual validations. You can sign up for a plan below or feel free to <a href="#" {{action "showSupport"}}>reach out</a> with any questions!
+              <strong>Your trial has expired.</strong> You're currently on the limited-use free plan, which provides {{subscriptionData.PLANS.[0].numWorkersTitle}} and {{subscriptionData.PLANS.[0].numDiffs}} snapshots {{snapshot-explanation-tooltip}}. You can sign up for a plan below or feel free to <a href="#" {{action "showSupport"}}>reach out</a> with any questions!
```
Don't think it's necessary here, personally.

timhaines replied:
Agree - I don't think it's necessary here.  Only on pricing page and billing page, where we're attaching dollars to them, in my mind..

PixelJanitor commented:
I prefer the info icon! Now that I see these side by side, i’m concerned a user will see the asterisk and go looking for the associated info lower on the page like that pattern represents throughout the rest of the interwebz.

fotinakis commented:
@cadeParade @timhaines I agree it should be present statically on the page as well and probably in replacement of the tooltip. If we have the tooltip, I agree it should be an info icon and not an asterisk as it doesn't look interactable. I think this should be one or the other, not both.

Added some other comments above.

cadeParade commented:
Ok, so reading through the comments on this PR, there are three terms proposed to take the place of the single previous term "visual validation".

I've updated the doc Mike made (https://docs.google.com/document/d/1EOkKSaHOuQegvuaWHArUZxTxbZDM6HqKMyNWc6qM-DQ/edit) with these three terms, and where I think they should be used. Please take a look and let me know if this sounds right.

paytonealexis commented:
LGTM

cadeParade commented:
Ok, I've updated based on comments above, and also updated the screenshots in the PR description for easy digestion of what changes have been made. (of course you can get the same information from Percy diffs, but you have to look through a bunch! Diff grouping!!)

Let me know!

timhaines commented:
@cadeParade  I like snapshot groups.  :)

![image](https://user-images.githubusercontent.com/19700/42542648-20889ff0-84a8-11e8-8a16-bea822a42de1.png)

^^ This needs a bit of refining.  
* `browser browser`.  
* We shouldn't mention tests here.  Tests only make up a subset of our customer's projects.
* Language that indicates that it's only for widths and browsers you have selected maybe.

cadeParade commented:
@timhaines 
How about this?
```
A snapshot is any screenshot plus diff, rendered in Percy. There will be a snapshot for each width and browser you have selected that's rendered across all of your projects.
```
